pkgs/by-name/sy/synchrony/package.nix

```diff
@@ -0,0 +1,89 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeBinaryWrapper,
+  nodejs,
+  pnpm,
+  nix-update-script,
+  fetchurl,
+  runCommand,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "synchrony";
+  version = "2.4.5";
+
+  src = fetchFromGitHub {
+    owner = "relative";
+    repo = "synchrony";
+    rev = finalAttrs.version;
+    hash = "sha256-nJ6H1SZAQCG6U3BPEPmm+BGQa8Af+Vb1E+Lv8lIqDBE=";
+  };
+
+  nativeBuildInputs = [
+    makeBinaryWrapper
+    nodejs
+    pnpm.configHook
+  ];
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-+hS4UK7sncCxv6o5Yl72AeY+LSGLnUTnKosAYB6QsP0=";
+  };
+
+  buildPhase = ''
+    runHook preBuild
+
+    pnpm run build
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    install -Dm755 dist/{index,cli}.js -t $out/share/synchrony
+    cp -r node_modules $out/share/synchrony
+    ln -s $out/share/synchrony/cli.js $out/bin/synchrony
+
+    runHook postInstall
+  '';
+
+  fixupPhase = ''
+    runHook preFixup
+
+    substituteInPlace $out/share/synchrony/cli.js \
+      --replace-fail "require('../')" "require('$out/share/synchrony')"
+    patchShebangs $out/share/synchrony/cli.js
+
+    runHook postFixup
+  '';
+
+  passthru = {
+    tests.deobfuscate =
+      let
+        obfuscated = fetchurl {
```

Comment:
Why not fetch both obfuscated and deobfuscated?

Reply:
There's no deobfuscated version provided by upstream — I had to run this on my local machine

Reply:
I suppose that doesn't stop you from uploading the deobfuscated file to gist.github.com though.

Reply:
> I suppose that doesn't stop you from uploading the deobfuscated file to gist.github.com though.

True, though that feels like more work than just sourcing from a file in-tree

Reply:
I would suggest to not commit the deobfuscate version at all and just check if it works.